src/libs/weather/src/SKY.cpp

```diff
@@ -65,207 +88,175 @@ void SKY::SetDevice()
+
+            vertices[idx] = CreateFogVertex(CVECTOR(R1 * fCos, h, R1 * fSin));
+
+            if (initialize)
+            {
+                if (y == kNumLevels - 1)
+                {
+                    *pIndices++ = static_cast<uint16_t>(kFogVertsNum - 1);
+                    *pIndices++ = static_cast<uint16_t>(idx);
+                    *pIndices++ = static_cast<uint16_t>((x == kNumAngles - 1) ? idx - (kNumAngles - 1) : idx + 1);
+                }
+                else
+                {
+                    *pIndices++ = static_cast<uint16_t>(idx);
+                    *pIndices++ = static_cast<uint16_t>((x == kNumAngles - 1) ? idx - (kNumAngles - 1) : idx + 1);
+                    *pIndices++ = static_cast<uint16_t>(idx + kNumAngles);
+
+                    *pIndices++ = static_cast<uint16_t>((x == kNumAngles - 1) ? idx - (kNumAngles - 1) : idx + 1);
+                    *pIndices++ = static_cast<uint16_t>(idx + kNumAngles);
+                    *pIndices++ = static_cast<uint16_t>((x == kNumAngles - 1) ? idx + 1 : idx + kNumAngles + 1);
+                }
+            }
 
             idx++;
         }
     }
-    vPos = CVECTOR(0.0f, R, 0.0f);
-    pVertBuf[idx].pos = vPos;
-    auto *pvData = core.Event(WEATHER_CALC_FOG_COLOR, "fff", vPos.x, vPos.y, vPos.z);
-    Assert(pvData);
-    pVertBuf[idx].diffuse = static_cast<uint32_t>(pvData->GetLong());
+    vertices[idx] = CreateFogVertex(CVECTOR(0.0f, kR, 0.0f));
+
+    auto *vertBuf = static_cast<FOGVERTEX *>(pRS->LockVertexBuffer(iFogVertsID));
+    memcpy(vertBuf, vertices, kFogVertsNum * sizeof(FOGVERTEX));
     pRS->UnLockVertexBuffer(iFogVertsID);
+
+    if (initialize)
+    {
+        auto *indBuf = static_cast<uint16_t *>(pRS->LockIndexBuffer(iFogIndexID));
+        memcpy(indBuf, indices, kFogTrgsNum * 2);
+        pRS->UnLockIndexBuffer(iFogIndexID);
+    }
 }
 
-void SKY::GenerateSky()
+void SKY::GenerateSky(const bool initialize)
 {
-    SKYVERTEX Verts[SKY_NUM_VERTEX];
-    long i;
-    SKYVERTEX v[8];
-
-    Release();
+    if (initialize)
+    {
+        Release();
+        LoadTextures();
+    }
 
-    v[0].pos = CVECTOR(-1.0f, 0.0f, -1.0f);
-    v[1].pos = CVECTOR(1.0f, 0.0f, -1.0f);
-    v[2].pos = CVECTOR(1.0f, 0.0f, 1.0f);
-    v[3].pos = CVECTOR(-1.0f, 0.0f, 1.0f);
-    v[4].pos = CVECTOR(-1.0f, 1.0f, -1.0f);
-    v[5].pos = CVECTOR(1.0f, 1.0f, -1.0f);
-    v[6].pos = CVECTOR(1.0f, 1.0f, 1.0f);
-    v[7].pos = CVECTOR(-1.0f, 1.0f, 1.0f);
+    if (iSkyVertsID < 0)
+    {
+        iSkyVertsID = pRS->CreateVertexBuffer(SKYVERTEX_FORMAT, SKY_NUM_VERTEX * sizeof(SKYVERTEX), D3DUSAGE_WRITEONLY);
+        
+    }
+    if (iSkyIndexID < 0)
+    {
+        iSkyIndexID = pRS->CreateIndexBuffer(20 * 3 * 2);
+    }
 
     const auto fpdelta = 1.0f / 1024.0f;
     const auto fpdx = 1.0f - fpdelta; //.25f - fpdelta;
     const auto fp0 = fpdelta;
-    auto fp1 = 0.25f + fpdelta;
-    auto fp2 = 0.5f + fpdelta;
-    auto fp3 = 0.75f + fpdelta;
+    //auto fp1 = 0.25f + fpdelta;
```

Comment:
is it on purpose?

Reply:
Yeah since those are not used. I think it's good practice for this project to comment out dead code instead of removing it when you have no idea why it is not used actually 😆 I'd remove this after revising all that math (one day 🤔 )